CosineDistance.lua

```diff
@@ -73,6 +73,10 @@ function CosineDistance:updateGradInput(input, gradOutput)
       not_batch = true
    end
 
+   if #self.gradInput ~= 2 then
```

Comment:
`clearState` is missing here, it should clean the [following fields](https://github.com/szagoruyko/nn/blob/clearState/CosineDistance.lua#L33-L38).

*Not important note*: I'd do in here for consistence something like
```lua
self.gradInput[1] = self.gradInput[1] or v1.new()
self.gradInput[2] = self.gradInput[2] or v1.new()
```